src/selection/merge.js

```diff
@@ -1,6 +1,7 @@
 import {Selection} from "./index.js";
 
-export default function(selection) {
+export default function(context) {
+  var selection = context.selection ? context.selection() : context;
```

Comment:
Oh wait, @Fil , I think this should fall through to the "invalid merge" message in the case that `context` is `undefined`. We would not want to end up with an error like "Cannot read property 'selection' of undefined."

```suggestion
  var selection = (context && context.selection) ? context.selection() : context;
```